apps/settings/js/media_storage.js

```diff
@@ -430,78 +431,46 @@ require([
 
   Volume.prototype.unmountSDCard = function volume_unmountSDCard(evt) {
     // Pop up a confirm window before unmount SD card.
-    var popup = document.getElementById('unmount-sdcard-dialog');
-    var cancelBtn = document.getElementById('unmount-sdcard-cancel-btn');
-    var okBtn = document.getElementById('unmount-sdcard-ok-btn');
-
     var self = this;
-    var confirmHandler = function() {
-      enablePopup(false);
-      // Unmount SD card
-      self.storages.sdcard.unmount();
-      self.isUnmounting = true;
-    };
-
-    var cancelHandler = function() {
-      enablePopup(false);
-    };
 
-    var enablePopup = function Vf_enablePopup(enabled) {
-      if (enabled) {
-        okBtn.addEventListener('click', confirmHandler);
-        cancelBtn.addEventListener('click', cancelHandler);
-        popup.hidden = false;
-      } else {
-        okBtn.removeEventListener('click', confirmHandler);
-        cancelBtn.removeEventListener('click', cancelHandler);
-        popup.hidden = true;
+    DialogService.confirm('unmount-sdcard-confirmation', {
+      submitButton: 'ok',
+      cancelButton: 'cancel'
+    }).then(function(result) {
+      if (result.type === 'submit') {
+        // Unmount SD card
+        self.storages.sdcard.unmount();
+        self.isUnmounting = true;
       }
-    };
-    enablePopup(true);
```

Comment:
I think we could just leave it, since enablePopup() only deals with the styles and the event binding of deleted elements. 
For result.type === 'cancel', the previous code use enablePopup(false) to make popup form hidden. 
AFAIK ,if we ignore it, 
https://github.com/mozilla-b2g/gaia/blob/59c8605/apps/settings/js/modules/dialog_manager.js#L204-L235  ,
will pass through the promise and resolve it directly, then dialog manager transit to next state to make it hidden.

It is my understanding, and please correct me if I'm wrong. Thanks a lot.